ui/apps/dashboard/src/app/(organization-active)/(dashboard)/settings/integrations/vercel/configure/[id]/page.tsx

```diff
@@ -127,11 +128,33 @@ export default function VercelConfigure() {
   // when enabling a project (e.g. the protection bypass secret)
   const areExtraSettingsVisible = project?.isEnabled && newEnablement !== 'enabled';
 
+  if (fetching) {
```

Comment:
Handle all data fetching states explicitly and check for data and project below. This removes the erroneous flash.